clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestState.java

```diff
@@ -132,6 +132,16 @@ public void onFailedAttempt(final long currentTimeMs) {
         this.numAttempts++;
     }
 
+    /**
+     * Set backoff and number of attempts to 0. This will ensure that the request is sent out
+     * again right away. Expected to be used when receiving errors responses that lead to a new
+     * request that can be sent without backing off (ex. member rejoining after fencing)
+     */
+    public void resetBackoff() {
```

Comment:
I just realized that we already have `reset` which is used for a similar purpose. Could we reuse it?

Reply:
yeap, that one was resetting the full state (not only the backoff), so initially I thought better to reset only what we need. But I totally agree that the "retries" after these errors can be seen as a fresh start, it does not harm to reset the full state, and it does simplify to reuse the single reset, done.